src/includes/class-wordlift-post-to-jsonld-converter.php

```diff
@@ -293,179 +295,96 @@ public function get_author( $author_id, &$references ) {
 	}
 
 	/**
-	 * Enrich the provided params array with publisher data, if available.
+	 * Set the publisher in the jsonld array by reference.
 	 *
-	 * @param array $params The parameters array.
+	 * @param array $jsonld     A reference to the jsonld array.
+	 * @param array $references A reference to the references array.
 	 *
-	 * @since 3.10.0
+	 * @since 3.53.0
 	 */
-	protected function set_publisher( &$params ) {
-		// @@todo add a &$references argument and push the publisher_id in here
-		// the `publisher` property should only reference the publisher e.g. { "@id": "https://data.example.org/dataset/publisher" }
-
-		// If the publisher id isn't set don't do anything.
-		$publisher_id = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
-		if ( empty( $publisher_id ) ) {
-			return;
-		}
-
-		// Get the post instance.
-		$post = get_post( $publisher_id );
-		if ( ! is_a( $post, '\WP_Post' ) ) {
-			// Publisher not found.
+	protected function set_publisher_by_reference( &$jsonld, &$references ) {
+		// If the publisher isn't set don't do anything.
+		if ( ! Wordlift_Publisher_Service::get_instance()->is_publisher_set() ) {
 			return;
 		}
 
-		// Get the item id.
-		$id = Wordlift_Entity_Service::get_instance()->get_uri( $publisher_id );
+		$publisher_id = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
+		$publisher_uri = Wordlift_Entity_Service::get_instance()->get_uri( $publisher_id );
 
-		// Get the type.
-		$type = $this->entity_type_service->get( $publisher_id );
+		$jsonld['publisher'] = array(
+			'@id' => $publisher_uri
+		);
 
-		// Get the name.
-		$name = $post->post_title;
+		$references[] = $publisher_id;
+	}
 
-		// Set the publisher data.
-		$params['publisher'] = array(
-			'@type' => $this->relative_to_context( $type['uri'] ),
-			'@id'   => $id,
-			'name'  => $name,
+	/**
+	 * Enrich the provided jsonld array with extra publisher data, if available.
+	 *
+	 * @param array $jsonld     A reference to the jsonld array.
+	 *
+	 * @since 3.10.0
+	 */
+	protected function expand_publisher_post( &$jsonld ) {
+		/**
+		 * Set the logo, only for http://schema.org/ + Organization, localBusiness, or onlineBusiness
+		 * as Person doesn't support the logo property.
+		 *
+		 * @see http://schema.org/logo.
+		 */
+		$organization_types = array(
+			'Organization',
+			'localBusiness',
+			'onlineBusiness'
```

Comment:
The condition about the logo is not the type of the entity, it is whether the entity post ID matches the configured publisher post ID.